xdsl/utils/deprecation.py

```diff
@@ -0,0 +1,20 @@
+from typing import Callable, ParamSpec, TypeVar
```

Comment:
Why this and not the deprecated package? Just to circumvent a dependency? Feels weird to reimplement stuff just for that, I guess Python packaging is somewhat broken?

Reply:
For 10/20 lines of code , I was thinking that  it's better to not use the package.
If we want more features, then maybe we should consider using it.

Reply:
Worth adding as a comment in the code?

Reply:
Done!